src/app/inventory/store/d2-item-factory.service.ts

```diff
@@ -923,15 +910,16 @@ export function D2ItemFactory(
 
   function buildMasterworkInfo(
     sockets: DimSockets,
-    statDefs: LazyDefinition<DestinyStatDefinition>
+    defs: D2ManifestDefinitions
   ): DimMasterwork | null {
-    const progress = _.find(_.pluck(sockets.sockets, 'masterworkProgress'), (mp) => mp >= 0);
-    const typeName = _.find(_.pluck(sockets.sockets, 'masterworkType'), (type) => type !== null);
-    const typeIcon = _.find(_.pluck(sockets.sockets, 'masterworkIcon'), (icon) => icon !== null);
-    const typeDesc = _.find(_.pluck(sockets.sockets, 'masterworkDesc'), (desc) => desc !== null);
-    const statHash = _.find(_.pluck(sockets.sockets, 'masterworkStat'), (ms) => ms > 0);
-    const statName = statDefs.get(statHash).displayProperties.name;
-    const statValue = _.find(_.pluck(sockets.sockets, 'masterworkValue'), (mv) => mv >= 0);
+    const socket = sockets.sockets[sockets.sockets.findIndex((socket) => socket.plugObjectives.length > 0)];
+    const progress = socket.plugObjectives[0].progress;
```

Comment:
Pulling out `socket.plugObjectives[0]` into a local variable will help reduce a lot of duplication.